src/modules/assertions/require_cc0.ts

```diff
@@ -0,0 +1,23 @@
+import { context, getOctokit } from "@actions/github";
+import { setFailed } from "@actions/core";
+import {
+  GITHUB_TOKEN,
+  GithubSelf,
+  PR,
+  Review
+} from "src/domain";
+import _ from "lodash";
+
+// BOT will fail if no CC0 detected in PR body
+// Just after: Verify correct environment and request context in main.ts, if no CC0 is detected, 
+// BOT will fail and comment: "error: No CC0 waiver detected" 
+
+export const  checkCC0 = () => {
+        
+	const github = getOctokit(GITHUB_TOKEN).rest;
+	if !(((context.payload?.pull_request?.body).endsWith("Copyright and related rights waived via [CC0](https://creativecommons.org/publicdomain/zero/1.0/).")) || ((context.payload?.pull_request?.body).endsWith("Copyright and related rights waived via [CC0](https://creativecommons.org/publicdomain/zero/1.0/).\n"))) {
```

Comment:
This, although it almost works (there's a syntax error), is a bit ugly.

How about the following?
```suggestion
  const copyrightRegex = /##Copyright\n*Copyright and related rights waived via \[CC0\]\(.+\)\.\n*$/g;
	if (context.payload?.pull_request?.body && copyrightRegex.test(context.payload?.pull_request?.body)) {
```
Regex (despite being somewhat harder to understand) is oftentimes better for this purpose as it avoids long and verbose lines like the above.

The above also permits linking to other instances of CC0 (e.g. the one in the root of the repository), and allows any number of newlines after the ##Copyright header and after the copyright text.

Also, how is this making sure that it's the EIP file that is being checked?

Reply:
Here's an explanation of the regex (it's simpler than you'd think): https://regexr.com/6l179

Reply:
I have used payload body.  is it not there where should be first mentioned ?

Reply:
I used endsWith as mentioned: CC0 must be at the end the body ? Question !

Thanks for the regex.

Reply:
There shouldn't be any spaces before the `## Copyright`, so we can remove that from the regex.  Why do you have a capture for the link rather than just including it?  If someone links to something besides `../License.md` that would be incorrect.

Reply:
Does TS regex support searching for "last thing in file"?  I know you can use $ for end of line, but I'm not sure if there is an end of file check?